lib/datasource/docker/index.spec.ts

```diff
@@ -255,6 +255,34 @@ describe(getName(), () => {
       expect(res).toBeNull();
       expect(httpMock.getTrace()).toMatchSnapshot();
     });
+    it('passes credentials to client', async () => {
+      httpMock
+        .scope(amazonUrl)
+        .get('/')
+        .reply(200, '', {
+          'www-authenticate': 'Basic realm="My Private Docker Registry Server"',
+        })
+        .get('/')
+        .reply(200)
+        .get('/node/manifests/some-tag')
+        .reply(200, '', { 'docker-content-digest': 'some-digest' });
+
+      await getDigest(
+        {
+          datasource: 'docker',
+          depName: '123456789.dkr.ecr.us-east-1.amazonaws.com/node',
+        },
+        'some-tag'
+      );
+
+      expect(AWS.ECR).toHaveBeenCalledWith({
+        credentials: {
+          accessKeyId: 'some-username',
+          secretAccessKey: 'some-password',
+        },
+        region: 'us-east-1',
+      });
```

Comment:
Add the nock trace expect.

Reply:
I deliberately left it out because the stuff passed to the constructor has no influence to the HTTP calls made (because `getAuthorizationToken` is mocked) and I am not a big fan of "useless" asserts. Or do you just want it for consistency/symmetry?

Reply:
Yes, i would like to have it, because we catch any unexpected http layer changes with that.

Reply:
You're the boss 😀 See b4496121390c8de30b76fcb6f7a3869acfb986f6